pkg/services/folder/folderimpl/folder.go

```diff
@@ -334,15 +337,48 @@ func (s *Service) GetChildren(ctx context.Context, q *folder.GetChildrenQuery) (
 		metrics.MFolderIDsServiceCount.WithLabelValues(metrics.Folder).Inc()
 		// nolint:staticcheck
 		f.ID = dashFolder.ID
+
+		hasAccess := parentHasAccess
```

Comment:
agree, but I think it's possible for a subfolder to have permission while the parent folder does not; so this change was for those cases
I will remove it though since it used to work as is (checking only the parent access - we have introduced the `shared with me` folder for the above case) and it does server the initial scope of this PR